gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/kubernetes/KubernetesSyncManager.java

```diff
@@ -56,11 +60,13 @@ protected void doStart() {
         log.debug("Starting kubernetes synchronization process");
         if (!distributedSyncService.isEnabled() || distributedSyncService.isPrimaryNode()) {
             watcherDisposable =
-                Flowable
-                    .fromIterable(synchronizers)
-                    .concatMapCompletable(kubernetesSynchronizer -> kubernetesSynchronizer.synchronize(environments))
-                    .doOnError(throwable -> log.error("An error occurred during kubernetes synchronization refresh. Restarting.", throwable)
-                    )
+                sync()
+                    .doOnComplete(() -> {
+                        log.debug("Moving to ready state as all resources have been synchronized");
+                        synced.set(true);
+                    })
+                    .andThen(watch())
+                    .doOnError(throwable -> log.error("An error occurred during Kubernetes synchronization. Restarting ...", throwable))
                     .retry()
```

Comment:
@guillaumelamirand recently [improved the sync process](https://github.com/gravitee-io/gravitee-api-management/blob/ee7e78cda5a3bba32af3115cc24452f34965f12a/gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/repository/DefaultSyncManager.java#L121) (the normal one) and changed the retry a bit. It could be a good idea to implement it the same here wdyt?